src/api/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -479,6 +483,42 @@ else if (damageReductionAmount <= ArmorMaterial.DIAMOND.getDamageReductionAmount
         return 5;
     }
 
+    /**
+     * Estimates the fishing rod tier from available durability and enchantment status.
+     *
+     * @param itemStack the tool to check.
+     * @return equivalent tool level.
+     */
+    private static int getFishingRodLevel(final ItemStack itemStack)
+    {
+        if (itemStack.getItem() == Items.FISHING_ROD)
+        {
+            return (1 + getMaxEnchantmentLevel(itemStack));
+        }
+        if (!itemStack.isDamageable())
+        {
+            return 1;
```

Comment:
How sure are we this is a good idea.
I would say if th rod is not damageable it is a max hut level feature.

Reply:
Fair.  I was thinking more for situations where it's the only craftable rod, but a) that should be done as an override to the vanilla one, b) an extreme edge case, and c) probably wouldn't be a bad place to make the fishing hut a higher-tier only choice anyway.